### PR TITLE
[NPU] add npu kernel for equal op

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

New features
### PR changes

OPs
### Describe

[NPU] add npu kernel for equal op

image

image

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/npu/test_compare_op_npu.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        x = np.random.uniform(1, 2, [11, 17]).astype(self.dtype)
+        y = x
+        out = x == y
+
+        self.inputs = {
+            'X': OpTest.np_dtype_to_fluid_dtype(x),
+            'Y': OpTest.np_dtype_to_fluid_dtype(y)
+        }
+        self.outputs = {'Out': out}
+
```
Comment (liym27) [inline python/paddle/fluid/tests/unittests/npu/test_compare_op_npu.py:77]: Add tests for `fp16` and `int32`

Comment (zhiqiu) [reply python/paddle/fluid/tests/unittests/npu/test_compare_op_npu.py:77]: done

File: paddle/fluid/operators/controlflow/compare_op_npu.cc [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (liym27) [inline paddle/fluid/operators/controlflow/compare_op_npu.cc]: 2016 -> 2021

Comment (zhiqiu) [reply paddle/fluid/operators/controlflow/compare_op_npu.cc]: done

**Review by liym27 — APPROVED** [review]

**Pull request merged**